### PR TITLE
refactor: make CI faster by installing pre-built scaffold binary

### PR DESCRIPTION
**Pull request by alexeagle**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -26,14 +26,14 @@ jobs:
...
             - uses: actions/checkout@v4
-            - uses: actions/setup-go@v5
-              with:
-                go-version: '>=1.22'
             - uses: bazel-contrib/setup-bazel@0.8.5
               with:
                 bazelisk-cache: true
                 repository-cache: true 
-            - run: go install github.com/hay-kot/scaffold@${{ matrix.scaffold-version }}
+            - uses: jaxxstorm/action-install-gh-release@25d5e2dd555cd74f1fab9ac1e6ea117acde2c0c4 # v1.12.0
```
Comment (jbedard) [inline .github/workflows/ci.yaml:33]: Why the hash if v1.12 is released?

Comment (jbedard) [inline .github/workflows/ci.yaml:33]: Have you used this anywhere else yet? Do we want to test it before putting it into a template and spreading it around?

Comment (alexeagle) [reply .github/workflows/ci.yaml:33]: this is a security posture we should adopt globally. The tag is mutable so anyone who gains write credentials to the repo (a leaked deploy token, say) could run arbitrary code on our CI

Comment (alexeagle) [reply .github/workflows/ci.yaml:33]: ah, this file is not part of the template. It's outside the `{{.ProjectSnake}}` folder so it's private only to us

**Review by jbedard — APPROVED** [review]

**Pull request merged**